services/business/istio_validations.go

```diff
@@ -21,17 +21,32 @@ func (in *IstioValidationsService) GetServiceValidations(namespace, service stri
 		return nil, err
 	}
 
+	objectCheckers := enabledCheckersFor(istioDetails)
+	objectCheckersCount := len(objectCheckers)
+
 	objectValidations := models.IstioValidations{}
+	validationReceiver := make(chan *models.IstioValidations, objectCheckersCount)
 
 	// Run checks for each IstioObject type
-	for _, objectChecker := range enabledCheckersFor(istioDetails) {
-		objectValidations = objectValidations.MergeValidations(objectChecker.Check())
+	for _, objectChecker := range objectCheckers {
+		go runObjectChecker(objectChecker, validationReceiver)
+	}
+
+	// Receive validations and merge them into one object
+	for validation := range validationReceiver {
+		objectValidations.MergeValidations(validation)
 	}
 
 	// Get groupal validations for same kind istio objects
 	return objectValidations, nil
 }
 
+func runObjectChecker(objectChecker ObjectChecker, validationChan chan *models.IstioValidations) {
+	defer close(validationChan)
```

Comment:
In some preliminar PR we closed all the channels but I think it is not really necessary.

Also, for this one liner methods (as you don't need the close) you can use the in-line go-routine.

It's a matter of style but that helps to reduce the calls instead to separate it to a different method.

Reply:
@lucasponce I can't make it work without the close statement. Tests got stuck and never finished.
Just found that for buffered channels you got to explicitly close it. Perhaps a little bit naive example:
https://tour.golang.org/concurrency/4

Do you recall where was that discussed?

Reply:
Clossing channel is fine.
I just remembered this one https://github.com/kiali/kiali/pull/30#pullrequestreview-97509265 .
In any case one or two liners code I guess is nicer to use the in-line goroutine.
But as commented, it's not a blocker from my side, more a discussion about style.